activitypub.php

```diff
@@ -50,19 +50,45 @@ function init() {
 	Rest\Webfinger::init();
 	Rest\Server::init();
 	Rest\Collection::init();
+}
+
+/**
+ * Initialize plugin.
+ */
+function plugin_init() {
+	define_constants();
+
+	add_action( 'init', array( __NAMESPACE__ . '\Migration', 'init' ) );
+	add_action( 'init', array( __NAMESPACE__ . '\Activitypub', 'init' ) );
+	add_action( 'init', array( __NAMESPACE__ . '\Activity_Dispatcher', 'init' ) );
+	add_action( 'init', array( __NAMESPACE__ . '\Collection\Followers', 'init' ) );
+
+	// Configure the REST API routes
+	add_action( 'init', __NAMESPACE__ . '\rest_init' );
 
-	Admin::init();
-	Hashtag::init();
-	Shortcodes::init();
-	Mention::init();
-	Health_Check::init();
-	Scheduler::init();
+	add_action( 'init', array( __NAMESPACE__ . '\Admin', 'init' ) );
+	add_action( 'init', array( __NAMESPACE__ . '\Hashtag', 'init' ) );
+	add_action( 'init', array( __NAMESPACE__ . '\Shortcodes', 'init' ) );
+	add_action( 'init', array( __NAMESPACE__ . '\Mention', 'init' ) );
+	add_action( 'init', array( __NAMESPACE__ . '\Health_Check', 'init' ) );
+	add_action( 'init', array( __NAMESPACE__ . '\Scheduler', 'init' ) );
 
 	if ( site_supports_blocks() ) {
-		Blocks::init();
+		add_action( 'init', array( __NAMESPACE__ . '\Blocks', 'init' ) );
+	}
+
+	// load NodeInfo endpoints only if blog is public
+	if ( \get_option( 'blog_public', 1 ) ) {
+		Rest\NodeInfo::init();
+	}
+
+	$debug_file = __DIR__ . '/includes/debug.php';
+	if ( \WP_DEBUG && file_exists( $debug_file ) && is_readable( $debug_file ) ) {
+		require_once $debug_file;
+		Debug::init();
 	}
 }
-\add_action( 'init', __NAMESPACE__ . '\init' );
+add_action( 'plugins_loaded', __NAMESPACE__ . '\plugin_init' );
```

Comment:
We changed this to `init` to fix some race conditions: https://github.com/Automattic/wordpress-activitypub/commit/ca5a3e24b14dec46d5060aa29336ec8bc0657b29